nanogpt/nanogpt_runner.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+The simple runner for nanoGPT. It is for demonstrating how to run the model exported to executorch runtime.
+
+*/
+
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <unordered_map>
+
+#include "basic_tokenizer.h"
+#include "managed_tensor.h"
+
+#include <executorch/extension/module/module.h>
+#include <executorch/extension/evalue_util/print_evalue.h>
+#include <executorch/runtime/core/exec_aten/exec_aten.h>
+#include <executorch/runtime/core/exec_aten/util/dim_order_util.h>
+#include <executorch/runtime/core/exec_aten/util/tensor_util.h>
+
+using namespace std;
+using namespace torch::executor;
+
+using SizesType = exec_aten::SizesType;
+using DimOrderType = exec_aten::DimOrderType;
+using StridesType = exec_aten::StridesType;
+
+
+vector<int64_t> generate(Module& llm_model, vector<int64_t>& tokens) {
+    // Convert the input tokens from a vector of int64_t to EValue.
+    // Evalue is a unified data type in the executorch runtime.
+    ManagedTensor tensor_tokens(tokens.data(), {1, 3}, ScalarType::Long);
+    vector<EValue> inputs = {tensor_tokens.get_aliasing_tensor()};
+
+    // Run the model given the Evalue inputs. The model will also return a sequence of EValues as output.
+    Result<vector<EValue>> output_res = llm_model.forward(inputs);
+
+    // Convert the output from EValue to a vector of int64_t.
+    Tensor output_tensor = output_res.get()[0].toTensor();
+    vector<int64_t> output_data(output_tensor.data_ptr<int64_t>(), output_tensor.data_ptr<int64_t>() + output_tensor.numel());
+
+    return output_data;
+}
+
+
+int main() {
+    // Load the input. Here we use "Hello world!" as sample input
+    string prompt = "Hello world!";
```

Comment:
Since we're parsing the prompt anyway, consider letting the user provide an optional prompt on the commandline -- it's more interactive and engaging.

Something like
```suggestion
int main(int argc, char** argv) {
    // Set up the input text that the model will respond to.
    string prompt = "Hello world!";
    if (argc > 1) {
        // Use the first commandline argument if provided.
        prompt = argv[1];
    }
```